src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3160,10 +3162,18 @@ void EmitLazy(RegexNode node)
                 SliceInputSpan();
                 if (doneLabel == originalDoneLabel)
                 {
+                    // Since the only reason we'd end up revisiting previous iterations of the lazy loop is if the child had backtracking constructs
+                    // we'd backtrack into, and the child doesn't, the whole loop is failed and done. If we successfully processed any iterations,
+                    // we thus need to pop all of the state we pushed onto the stack for those iterations, as we're exiting out to the parent who
+                    // will expect the stack to be cleared of any child state.
+                    int entriesPerIteration = 3 + (expressionHasCaptures ? 1 : 0);
```

Comment:
I suppose the number 3 refers to line 3093 above?

Reply:
Yup. I have a subsequent cleanup I've been noodling on and as part of that I move this up to be right next to it, assuming I go ahead with it.